src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -25,147 +25,108 @@
+                            + " configured in Jenkins global configuration.");
+            return;
         }
-    }
-
-    @CheckForNull
-    private static String extractRevision(Run<?, ?> build) {
-        String revision = null;
-        BuildData gitBuildData = build.getAction(BuildData.class);
-        if (gitBuildData != null) {
-            Revision lastBuiltRevision = gitBuildData.getLastBuiltRevision();
-            if (lastBuiltRevision != null) {
-                revision = lastBuiltRevision.getSha1String();
-            }
+        String key = build.getParent().getFullName(); // use the job full name as the key for the status
+        String name = build.getDisplayName(); // use the build number as the display name of the status
+        BitbucketBuildStatus status;
+        Result result = build.getResult();
+        if (Result.SUCCESS.equals(result)) {
+            status = new BitbucketBuildStatus(hash, "This commit looks good", "SUCCESSFUL", url, key, name);
+        } else if (Result.UNSTABLE.equals(result)) {
+            status = new BitbucketBuildStatus(hash, "This commit has test failures", "FAILED", url, key, name);
+        } else if (Result.FAILURE.equals(result)) {
+            status = new BitbucketBuildStatus(hash, "There was a failure building this commit", "FAILED", url, key,
+                    name);
+        } else if (result != null) { // ABORTED etc.
+            status = new BitbucketBuildStatus(hash, "Something is wrong with the build of this commit", "FAILED", url,
+                    key, name);
         } else {
-            MercurialTagAction action = build.getAction(MercurialTagAction.class);
-            if (action != null) {
-                revision = action.getId();
-            }
+            status = new BitbucketBuildStatus(hash, "The tests have started...", "INPROGRESS", url, key, name);
+        }
+        new BitbucketChangesetCommentNotifier(bitbucket).buildStatus(status);
+        if (result != null) {
+            listener.getLogger().println("[Bitbucket] Build result notified");
         }
-        return revision;
     }
 
-    private static void createPullRequestCommitStatus(Run<?,?> build, TaskListener listener, BitbucketApi bitbucket)
+    private static void sendNotifications(Run<?, ?> build, TaskListener listener)
             throws IOException, InterruptedException {
-        createBuildCommitStatus(build, listener, bitbucket);
-    }
-
-    private static BitbucketNotifier getNotifier(BitbucketApi bitbucket) {
-        return new BitbucketChangesetCommentNotifier(bitbucket);
-    }
+        final SCMSource s = SCMSource.SourceByItem.findSource(build.getParent());
+        if (!(s instanceof BitbucketSCMSource)) {
+            return;
+        }
+        BitbucketSCMSource source = (BitbucketSCMSource) s;
+        if (new BitbucketSCMSourceContext(null, SCMHeadObserver.none())
+                .withTraits(source.getTraits())
+                .notificationsDisabled()) {
```

Comment:
Nope. I do not see the value in there being a notification disabling trait in the core plugin. But if somebody wants to write such a trait in an extension plugin they need this to disable the notifications... given that there are a number of people seeking just that I thought it worth the minor YAGNI to add this